src/robot/utils/robottypes.py

```diff
@@ -12,9 +12,26 @@
 #  See the License for the specific language governing permissions and
 #  limitations under the License.
 
+import sys
+
+PY3 = sys.version_info[0] == 3
+
+from .platform import IRONPYTHON
 from collections import Mapping
-from UserDict import UserDict
-from UserString import UserString
+if PY3:
+    from collections import UserList, UserDict, UserString
+else:
+    from UserList import UserList
+    from UserDict import UserDict
+    from UserString import UserString, MutableString
+if PY3:
+    long = int
+    bytes = bytes
+    unicode = str
+else:
+    long = long
+    bytes = str if not IRONPYTHON else bytes
+    unicode = unicode
```

Comment:
`long` and `unicode` are actually used in the code. And I often do `if PY3: long = int` and `from six import text_type as unicode` on demand in #1506 to make these cases work. I also changed some `str` uses to `bytes` for compatibility there.